prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -110,40 +109,35 @@ class ReportFunction(
         if (senderName.isBlank())
             return HttpUtilities.bad(request, "Expected a '$CLIENT_PARAMETER' query parameter")
 
-        // Sender should eventually be obtained directly from who is authenticated
-        val sender = workflowEngine.settings.findSender(senderName)
-            ?: return HttpUtilities.bad(request, "'$CLIENT_PARAMETER:$senderName': unknown sender")
-
-        val authenticationStrategy = AuthenticationStrategy.authStrategy(
-            request.headers["authentication-type"],
-            PrincipalLevel.USER,
-            workflowEngine.db
-        )
-
+        actionHistory.trackActionParams(request)
         try {
-            actionHistory.trackActionParams(request)
-
-            if (authenticationStrategy is OktaAuthentication) {
-                // The report is coming from a sender that is using Okta, so set "oktaSender" to true
-                return authenticationStrategy.checkAccess(request, sender.organizationName, true, actionHistory) {
-                    return@checkAccess processRequest(request, sender)
-                }
-            }
-
-            if (authenticationStrategy is TokenAuthentication) {
-                val claims = authenticationStrategy.checkAccessToken(request, "${sender.fullName}.report")
-                    ?: return HttpUtilities.unauthorizedResponse(request)
-                logger.info("Claims for ${claims["sub"]} validated.  Beginning ingestReport.")
-                return processRequest(request, sender)
+            val claims = AuthenticationStrategy.authenticate(request)
```

Comment:
This `authenticate`  is the new method that hides the gory details of doing both Okta and Server2server authentication.    And yes, this does authentication only.

Note the improvement that its now called immediately, with no prior database calls, unlike the old code.  This is a DoS mitigation.